src/vulkan/mod.rs

```diff
@@ -17,14 +17,14 @@ use crate::{
 
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub enum AllocationScheme {
-    /// This allocation will be for a buffer, and it will be driver managed using vulkans DedicatedAllocation feature.
-    /// The driver will be able to perform optimizations in some cases with this type of allocation.
-    DedicatedBuffer { buffer: vk::Buffer },
-    /// This allocation will be for a image, and it will be driver managed using vulkans DedicatedAllocation feature.
-    /// The driver will be able to perform optimizations in some cases with this type of allocation.
-    DedicatedImage { image: vk::Image },
-    /// This allocation will be managed by the GPU allocator.
-    /// It is possible that the allocation will reside in an allocation block shared with other resources.
+    /// Perform a dedicated, driver-managed allocation for the given buffer, allowing
+    /// it to perform optimizations on this type of allocation.
+    DedicatedBuffer(vk::Buffer),
+    /// Perform a dedicated, driver-managed allocation for the given image, allowing
+    /// it to perform optimizations on this type of allocation.
+    DedicatedImage(vk::Image),
+    /// The memory for this resource will be allocated and managed by gpu-allocator.
+    /// It is possible that the allocation will reside in memory shared with other resources.
```

Comment:
```suggestion
    /// It is possible that the allocation will reside in [`vk::Memory`] shared with other resources.
```

We have `vk` in scope right? Otherwise "allocation block" is more clear :)

Reply:
Clippy was complaining that there was no `Memory` defined in `vk`. I suppose allocation block works too but I have no strong preference towards either.

Reply:
`DeviceMemory` maybe? My mind might be fuzzy on the exact type naming.